src/main.rs

```diff
@@ -52,18 +61,19 @@ async fn main() -> Result<()> {
     match matches.subcommand() {
         Some(("reconstruct", sub_matches)) => match sub_matches.subcommand() {
             Some(("l1", args)) => {
+                // TODO: Use start_block from snapshot.
                 let start_block = args.get_one::<u64>("start-block").expect("required");
                 let block_step = args.get_one::<u64>("block-step").expect("required");
                 let http_url = args.get_one::<String>("http-url").expect("required");
                 println!("reconstruct from L1, starting from block number {}, processing {} blocks at a time", start_block, block_step);
 
+                // TODO: This should be an env variable / CLI argument.
+                let db_dir = env::current_dir()?.join("db");
+
                 let fetcher = L1Fetcher::new(http_url)?;
-                let (tx, mut rx) = mpsc::channel::<Vec<CommitBlockInfoV1>>(5);
-                tokio::spawn(async move {
-                    while let Some(blks) = rx.recv().await {
-                        blks.iter().for_each(|x| println!("{:?}", x));
-                    }
-                });
+                let processor = TreeProcessor::new(&db_dir)?;
+                let (tx, rx) = mpsc::channel::<Vec<CommitBlockInfoV1>>(5);
+                processor.run(rx);
```

Comment:
The returned `JoinHandle` is not captured nor "awaited" here - how does this work?

Reply:
This is the same behavior that we had before, we're just spawning the task inside this function instead. The thing with `tokio` futures is that they will start executing as soon as they're spawned, `await` will just wait until that task is done.